pwndbg/gdblib/tui/context.py

```diff
@@ -187,19 +187,23 @@ def _ansi_substr(self, line: str, start_char: int, end_char: int) -> str:
                 char_count += 1
             colored_idx += 1
         return (
-            (
-                ansi_escape_before_start
-                + line[colored_start_idx:colored_end_idx]
-                + ansi_escape_after_end
-                # Workaround for display bug in GDB TUI ANSI escape sequence translation.
-                # Only resetting the foreground or background colors results in
-                # generating wrong colors in the TUI.
-                # The workaround is to avoid the sequences that only reset the colors
-                # and replace them with a full reset sequence.
-                # It resets other styling attributes like bold too but it's better
-                # than having wrong colors.
-                # https://github.com/pwndbg/pwndbg/issues/2654
-            )
+            ansi_escape_before_start
+            + line[colored_start_idx:colored_end_idx]
+            + ansi_escape_after_end
+        )
+
+    # Workaround for display bug in GDB TUI ANSI escape sequence translation.
+    # Only resetting the foreground or background colors results in
+    # generating wrong colors in the TUI.
+    # The workaround is to avoid the sequences that only reset the colors
+    # and replace them with a full reset sequence.
+    # It resets other styling attributes like bold too but it's better
+    # than having wrong colors.
+    # https://github.com/pwndbg/pwndbg/issues/2654
+    if tuple(map(int, gdb.VERSION.split(".")[:2])) < (16, 3):
```

Comment:
@peace-maker btw could we use https://github.com/pwndbg/pwndbg/blob/eac203cf10fcb25d3884625e87230ab612190612/pwndbg/gdblib/__init__.py#L15 here?

Reply:
Yes! I was looking for something like this but didn't find it.